parser-typechecker/src/Unison/CommandLine/OutputMessages.hs

```diff
@@ -797,6 +801,29 @@ displayDefinitions' ppe types terms = P.syntaxToColor $ P.sep "\n\n" (prettyType
     <> P.newline
     <> tip "You might need to repair the codebase manually."
 
+displayRendered :: Maybe FilePath -> Pretty -> IO Pretty
+displayRendered outputLoc pp = 
+  maybe (pure pp) scratchAndDisplay outputLoc
+  where
+  scratchAndDisplay path = do
+    path' <- canonicalizePath path
+    prependToFile pp path'
+    pure (message pp path')
+    where
+    prependToFile pp path = do
+      existingContents <- do
+        exists <- doesFileExist path
+        if exists then readFile path
+        else pure ""
+      writeFile path . Text.pack . P.toPlain 80 $
+        P.lines [ pp, "", P.text existingContents ]
```

Comment:
Would it make sense to share this definition with the copy that's local to `displayDefinitions`?  Maybe `scratchAndDisplay` too?

Reply:
(It's fine if you don't want to.)

Reply:
didn't end up doing this one...